### PR TITLE
Add usage examples for Debug2Format, Display2Format

### PR DESCRIPTION
**Pull request by Lotterleben**
Also change some wording to put the most important info first

(drive-by improvement because I had to use these adapters and noticed they were missing some docs that I would've found helpful)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -741,8 +753,25 @@ impl<T: fmt::Debug + ?Sized> Format for Debug2Format<'_, T> {
...
 /// `defmt::Format` over `Display` whenever possible.
 ///
-/// Note that this always uses `{}` to format the contained value, meaning that any provided defmt
-/// display hints will be ignored.
+/// example usage:
+///
+/// ```rust
+/// # struct ExpensiveThing();
+///
+/// # impl core::fmt::Display for ExpensiveThing {
```
Comment (Urhengulas) [inline src/lib.rs:761]: I guess you forgot the `#` :) Else there is a newline.

```suggestion
/// # struct ExpensiveThing();
/// #
/// # impl core::fmt::Display for ExpensiveThing {
```

File: src/lib.rs [hunk 2]
```diff
@@ -722,8 +722,20 @@ fn default_panic() -> ! {
 /// This adapter disables compression and uses the `core::fmt` code on-device! You should prefer
 /// `defmt::Format` over `Debug` whenever possible.
 ///
-/// Note that this always uses `{:?}` to format the contained value, meaning that any provided defmt
-/// display hints will be ignored.
+/// example usage:
```
Comment (jonas-schievink) [inline src/lib.rs:725]:
```suggestion
/// # Examples
```

File: src/lib.rs [hunk 3]
```diff
@@ -741,8 +753,25 @@ impl<T: fmt::Debug + ?Sized> Format for Debug2Format<'_, T> {
 /// This adapter disables compression and uses the `core::fmt` code on-device! You should prefer
 /// `defmt::Format` over `Display` whenever possible.
 ///
-/// Note that this always uses `{}` to format the contained value, meaning that any provided defmt
-/// display hints will be ignored.
+/// example usage:
```
Comment (jonas-schievink) [inline src/lib.rs:756]:
```suggestion
/// # Examples
```

**Pull request merged**